<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Emily">

<title>Lab Meeting April 25th</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Lab meeting Put together_files/libs/clipboard/clipboard.min.js"></script>
<script src="Lab meeting Put together_files/libs/quarto-html/quarto.js"></script>
<script src="Lab meeting Put together_files/libs/quarto-html/popper.min.js"></script>
<script src="Lab meeting Put together_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Lab meeting Put together_files/libs/quarto-html/anchor.min.js"></script>
<link href="Lab meeting Put together_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Lab meeting Put together_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Lab meeting Put together_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Lab meeting Put together_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Lab meeting Put together_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="Lab meeting Put together_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="Lab meeting Put together_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Lab Meeting April 25th</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Emily </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="load-packages-and-set-directories" class="level2">
<h2 class="anchored" data-anchor-id="load-packages-and-set-directories">Load Packages and Set Directories</h2>
</section>
<section id="set-directories" class="level2">
<h2 class="anchored" data-anchor-id="set-directories">Set Directories</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># clear out working space </span><span class="al">###</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#load directories ####</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> <span class="st">"/Volumes/IPHY/ADORLab/HEI Study/Master Datasets/Metadata/long/"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="st">"/Volumes/IPHY/ADORLab/__Users/emye7956/MM/code_review/metadata"</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>d3 <span class="ot">&lt;-</span> <span class="st">"/Volumes/IPHY/ADORLab/HEI Study/16S Data/MM 16S Processed Files for Mom and Infants 05Jan21/alpha diversity/"</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># read in meta data ####</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># meta_og &lt;- read.csv(paste0(d1,"3_mothersMilk_metadata_timepointsAsRows_updated051022_Temporary24mDiet.csv"))</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#read in meta data with alpha diversity </span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>meta_alpha <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/Volumes/IPHY/ADORLab/__Users/emye7956/MM/code_review/input_files/a_meta.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="missing-variables-counts-and-summaries" class="level2">
<h2 class="anchored" data-anchor-id="missing-variables-counts-and-summaries">Missing Variables Counts and Summaries</h2>
<p>Getting an idea of all variables (sum1), cognitive variables (sum2), and socio-economic/biological variables (sum3).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create summary tables</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>meta_alpha6 <span class="ot">&lt;-</span> meta_alpha <span class="sc">%&gt;%</span> </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"6"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#all</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>sum1 <span class="ot">&lt;-</span> meta_alpha6 <span class="sc">%&gt;%</span> </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  summary <span class="sc">%&gt;%</span> </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xtable</span>()</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># of just cognitive variables:</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>sum2 <span class="ot">&lt;-</span> meta_alpha6 <span class="sc">%&gt;%</span> </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="at">select =</span> <span class="fu">c</span>(bsid_cog_cs,bsid_mot_cs, bsid_lang_cs,bsid_mot_fm_ss,bsid_mot_gm_ss,bsid_ab_cs,bsid_se_cs, cog_low2high)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xtable</span>()</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># of other variables:</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>sum3 <span class="ot">&lt;-</span> meta_alpha6 <span class="sc">%&gt;%</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="at">select =</span> <span class="fu">c</span>(<span class="co">#socio-economic variables</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                <span class="st">'SES_index_final'</span>,<span class="st">'moms_ed_level'</span>,<span class="st">'martial_status'</span>,<span class="st">'pets'</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                <span class="co">#biological variables</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>                <span class="st">'baby_birthweight_kg'</span>,<span class="st">'prepreg_bmi_kgm2'</span>,<span class="st">'mother_age'</span>, </span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>                <span class="st">'father_age'</span>,<span class="st">'inf_antibiotics'</span>,<span class="st">'breastmilk_per_day'</span>,</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>                <span class="st">'mode_of_delivery'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xtable</span>()</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="co"># print table as html</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">htmlTable</span>(sum2)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>&lt;table class='gmisc_table' style='border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;' &gt;
&lt;thead&gt;
&lt;tr&gt;&lt;th style='border-bottom: 1px solid grey; border-top: 2px solid grey;'&gt;&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt; bsid_cog_cs&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt; bsid_mot_cs&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt; bsid_lang_cs&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt;bsid_mot_fm_ss&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt;bsid_mot_gm_ss&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt;  bsid_ab_cs&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt;  bsid_se_cs&lt;/th&gt;
&lt;th style='font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;'&gt;cog_low2high&lt;/th&gt;
&lt;/tr&gt;
&lt;/thead&gt;
&lt;tbody&gt;
&lt;tr&gt;
&lt;td style='text-align: left;'&gt;X&lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 40.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 61.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 47.0  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 4.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 3.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 50.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Min.   : 55.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Length:209        &lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td style='text-align: left;'&gt;X.1&lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 90.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 88.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 77.0  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 8.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 7.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 84.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;1st Qu.: 90.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Class :character  &lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td style='text-align: left;'&gt;X.2&lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median : 95.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median : 94.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median : 83.0  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median :10.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median : 8.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median : 96.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Median :100.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mode  :character  &lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td style='text-align: left;'&gt;X.3&lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 93.58  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 94.32  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 84.2  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 9.894  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 8.169  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 96.04  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Mean   : 99.75  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td style='text-align: left;'&gt;X.4&lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.:100.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.:100.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.: 91.0  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.:11.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.: 9.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.:108.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;3rd Qu.:110.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td style='text-align: left;'&gt;X.5&lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :130.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :133.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :115.0  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :18.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :16.000  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :136.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;Max.   :140.00  &lt;/td&gt;
&lt;td style='text-align: center;'&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: left;'&gt;X.6&lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :30  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :32  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :33  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :30  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :32  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :30  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;NA's   :30  &lt;/td&gt;
&lt;td style='border-bottom: 2px solid grey; text-align: center;'&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;</code></pre>
</div>
</div>
</section>
<section id="removing-infants-with-incomplete-cognitive-assessments" class="level2">
<h2 class="anchored" data-anchor-id="removing-infants-with-incomplete-cognitive-assessments">Removing infants with incomplete cognitive assessments</h2>
<p>alpha meta is the complete data set with all infants. I filtered for those with complete cognitive scores, and without notes to suggest problems during testing. This included the following dyad-id’s:</p>
<p>- 101: missing language items.</p>
<p>- 176: Really shy, scared of everyone. Cried most of the time</p>
<p>- 180: Child was tired difficulty completing language portion</p>
<p>- 205: Child uncooperative, did not attempt anything not interested in. Would throw anything not interested in.</p>
<p>- 209: Mom is concerned about child. Child does not talk and is angry a lot</p>
<p>Removing those above, there were - 17 NA’s, - 21 incomplete cases - 5 suspicious cases - (171 - 5) = 166 complete cases</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># counting how many infants are complete </span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>c <span class="ot">&lt;-</span> meta_alpha <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"6"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(bayleys_test_bsid3_complete <span class="sc">==</span> <span class="st">"2"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>c</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  bayleys_test_bsid3_complete == "2"   n
1                              FALSE  21
2                               TRUE 171
3                                 NA  17</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>amc1 <span class="ot">&lt;-</span> meta_alpha <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bayleys_test_bsid3_complete <span class="sc">==</span> <span class="st">"2"</span>) <span class="co">#keeping only complete cases</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>amc <span class="ot">&lt;-</span> amc1 <span class="sc">%&gt;%</span>  <span class="co">#removing special cases </span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>dyad_id <span class="sc">==</span> <span class="st">"101"</span>,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span>dyad_id <span class="sc">==</span> <span class="st">"176"</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span>dyad_id <span class="sc">==</span> <span class="st">"180"</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span>dyad_id <span class="sc">==</span> <span class="st">"205"</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span>dyad_id <span class="sc">==</span> <span class="st">"209"</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#counting NA values </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co">#colSums(is.na(amc %&gt;% filter(timepoint == "6")))</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#colSums(is.na(amc %&gt;% filter(timepoint == "12")))</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#colSums(is.na(amc %&gt;% filter(timepoint == "24"))) # only have alpha diversity data for 6 and 12 months </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="check-for-outliers-in-cognitive-variables" class="level2">
<h2 class="anchored" data-anchor-id="check-for-outliers-in-cognitive-variables">Check for Outliers in cognitive variables</h2>
<p>Using the outlier package.</p>
<p>- group: (Optional) string indicating name of the grouping variable. If filled all further steps are done within-group.</p>
<p>- outlier: how flag outliers. ‘z’ flags values outside ± 3.28 SD range. ‘iqr’ flags values outside 1.5 * IQR range.</p>
<ul>
<li>treat:how treat outliers. ‘trim’ set them to NA. ‘win’ replace them by max/min. ‘replace’ does mean-replacement.</li>
</ul>
<p>Leaving them in for now. But worth exploring the results without.</p>
<p>Bayleys classifies &lt;70 as severe mental developmental impairment</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(amc<span class="sc">$</span>bsid_cog_cs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cog_vars &lt;- amc %&gt;% </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#subset(select = c('bsid_cog_cs','bsid_lang_cs','bsid_lang_cs','bsid_mot_cs','bsid_mot_fm_ss',</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                'bsid_mot_gm_ss','bsid_se_cs', 'bsid_ab_cs')) #variables to look up for.</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co">#new_cog_vars &lt;- outlyr(amc, cog_vars, group = 'baby_gender', outlier = 'z', treat = 'trim')  </span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#colSums(is.na(new_cog_vars)) #showing the number of NAs/outliers introduced</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="check-for-outliers-in-alpha-diversity-variables" class="level2">
<h2 class="anchored" data-anchor-id="check-for-outliers-in-alpha-diversity-variables">Check for Outliers in alpha diversity variables</h2>
<p>at 6 months, there were 72,71 and 72 outliers for gut richness, Shannon entropy, faith pd. There were apparently 107 outliers identified across all 3 metrics using the ± 3.28 SD methods and 114,112,and 114 identified using the iqr method. Seems a bit too high looking at the plots to remove</p>
<p>Leaving them in for now. But worth exploring the results without.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>amc6 <span class="ot">&lt;-</span> amc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"6"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>amc12 <span class="ot">&lt;-</span> amc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"12"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(amc6<span class="sc">$</span>gut_richness) <span class="co">#participant 83</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(amc6<span class="sc">$</span>faith_pd) <span class="co">#176</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(amc6<span class="sc">$</span>shannon_entropy)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-5-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>alpha_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'gut_richness'</span>,<span class="st">'shannon_entropy'</span>,<span class="st">'faith_pd'</span>) <span class="co"># changing to outliers to NA </span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co">#new_a_vars6 &lt;- outlyr(amc6, alpha_vars, group = 'baby_gender', outlier = 'z', treat = 'trim')</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#new_a_vars12 &lt;- outlyr(amc12, alpha_vars, group = 'baby_gender', outlier = 'iqr', treat = 'trim')</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#colSums(is.na(new_a_vars6))</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#colSums(is.na(new_a_vars12))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="summary-table-for-breast-feeding-and-alpha-diversity" class="level2">
<h2 class="anchored" data-anchor-id="summary-table-for-breast-feeding-and-alpha-diversity">Summary table for breast feeding and alpha diversity</h2>
<p>As breast feedings per day go down (indicating a switch to a variety of solid foods), alpha diversity increases.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mean breast milk by timepoint </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>meta_alpha <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(timepoint) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">Mean breastmilk per day</span><span class="st">`</span><span class="ot">=</span> <span class="fu">mean</span>(breastmilk_per_day, <span class="at">na.rm =</span> T),</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Mean Faith pd</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">mean</span>(faith_pd, <span class="at">na.rm =</span> T),</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Mean Shannon</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">mean</span>(shannon_entropy, <span class="at">na.rm =</span> T), </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Mean Gut Richness</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">mean</span>(gut_richness, <span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">kable_classic</span>(<span class="at">full_width =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:right;"> timepoint </th>
   <th style="text-align:right;"> Mean breastmilk per day </th>
   <th style="text-align:right;"> Mean Faith pd </th>
   <th style="text-align:right;"> Mean Shannon </th>
   <th style="text-align:right;"> Mean Gut Richness </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 8.219048 </td>
   <td style="text-align:right;"> 5.93055 </td>
   <td style="text-align:right;"> 2.203338 </td>
   <td style="text-align:right;"> 32.96575 </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 6 </td>
   <td style="text-align:right;"> 4.034826 </td>
   <td style="text-align:right;"> 7.81246 </td>
   <td style="text-align:right;"> 2.840534 </td>
   <td style="text-align:right;"> 52.33929 </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 12 </td>
   <td style="text-align:right;"> 2.571429 </td>
   <td style="text-align:right;"> 10.45680 </td>
   <td style="text-align:right;"> 3.443094 </td>
   <td style="text-align:right;"> 89.39130 </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:right;"> 2.348387 </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 24 </td>
   <td style="text-align:right;"> 2.027624 </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
   <td style="text-align:right;"> NaN </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
<section id="table-one-descriptive-stats" class="level2">
<h2 class="anchored" data-anchor-id="table-one-descriptive-stats">Table One Descriptive Stats</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>meta_alpha6 <span class="ot">&lt;-</span> meta_alpha <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"6"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>vars<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'timepoint'</span>,<span class="st">'bsid_cog_cs'</span>,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="st">'SES_index_final'</span>,<span class="st">'moms_ed_level'</span>,<span class="st">'martial_status'</span>,<span class="st">'pets'</span>,<span class="st">'baby_gender'</span>,<span class="st">'baby_birthweight_kg'</span>,<span class="st">'prepreg_bmi_kgm2'</span>,<span class="st">'mother_age'</span>, <span class="st">'father_age'</span>,<span class="st">'inf_antibiotics'</span>,<span class="st">'breastmilk_per_day'</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>,<span class="st">'gestational_age_category'</span>,<span class="st">'mode_of_delivery'</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co">#alpha diversity variables </span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a> ,<span class="st">'shannon_entropy'</span>, <span class="st">'faith_pd'</span>,<span class="st">'gut_richness'</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>cont_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'bsid_cog_cs'</span>, <span class="st">'baby_birthweight_kg'</span>,<span class="st">'prepreg_bmi_kgm2'</span>,<span class="st">'mother_age'</span>, <span class="st">'father_age'</span>,<span class="st">'SES_index_final'</span>)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>cat_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'timepoint'</span>,<span class="st">'moms_ed_level'</span>,<span class="st">'martial_status'</span>,<span class="st">'pets'</span>,<span class="st">'baby_gender'</span>,<span class="st">'inf_antibiotics'</span>,<span class="st">'breastmilk_per_day'</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>,<span class="st">'gestational_age_category'</span>,<span class="st">'mode_of_delivery'</span>)</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="fu">CreateTableOne</span>(<span class="at">vars =</span> vars, <span class="at">data =</span> meta_alpha, <span class="at">factorVars =</span> cat_vars, <span class="at">addOverall =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                 
                                  Overall      
  n                                1052        
  timepoint (%)                                
     1                              219 (20.8) 
     6                              209 (19.9) 
     12                             177 (16.8) 
     18                             162 (15.4) 
     24                             187 (17.8) 
     36                              98 ( 9.3) 
  bsid_cog_cs (mean (SD))         93.82 (12.46)
  SES_index_final (mean (SD))     26.91 (12.28)
  moms_ed_level (%)                            
     1                               72 ( 7.8) 
     2                               38 ( 4.1) 
     3                              157 (17.0) 
     4                              245 (26.5) 
     5                              242 (26.2) 
     6                              124 (13.4) 
     7                               45 ( 4.9) 
  martial_status (%)                           
     1                              249 (27.0) 
     2                              387 (41.9) 
     3                                5 ( 0.5) 
     4                                9 ( 1.0) 
     5                                1 ( 0.1) 
     6                              254 (27.5) 
     7                               18 ( 2.0) 
  pets = 1 (%)                      271 (29.4) 
  baby_gender = Male (%)            485 (46.1) 
  baby_birthweight_kg (mean (SD))  3.39 (0.41) 
  prepreg_bmi_kgm2 (mean (SD))    28.49 (5.89) 
  mother_age (mean (SD))          29.74 (6.15) 
  father_age (mean (SD))          32.57 (7.70) 
  inf_antibiotics = 1 (%)            36 ( 3.9) 
  breastmilk_per_day (%)                       
     0                              359 (38.9) 
     1                               34 ( 3.7) 
     2                               33 ( 3.6) 
     3                               44 ( 4.8) 
     4                               54 ( 5.9) 
     5                               60 ( 6.5) 
     6                               45 ( 4.9) 
     7                               39 ( 4.2) 
     8                               38 ( 4.1) 
     9                               37 ( 4.0) 
     10                             179 (19.4) 
  gestational_age_category (%)                 
                                     36 ( 3.4) 
     &lt;38                             36 ( 3.4) 
     &gt;42                             26 ( 2.5) 
     38-40                          198 (18.8) 
     40                             538 (51.1) 
     40-42                          218 (20.7) 
  mode_of_delivery = 2 (%)          249 (24.1) 
  shannon_entropy (mean (SD))      2.63 (0.92) 
  faith_pd (mean (SD))             7.31 (2.63) 
  gut_richness (mean (SD))        48.64 (30.68)</code></pre>
</div>
</div>
</section>
<section id="dag-lm-using-regression-to-assess-descriptive-stats-at-t6m" class="level2">
<h2 class="anchored" data-anchor-id="dag-lm-using-regression-to-assess-descriptive-stats-at-t6m">DAG lm: Using regression to assess descriptive stats at t6m</h2>
<p>lm1: Original model I was working with</p>
<p>lm2: DAG 1: SES and moms education were significant predictors. gender was nearly</p>
<p>lm3: DAG 2: SES and gender only</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> SES_index_final <span class="sc">+</span> breastmilk_per_day <span class="sc">+</span> gestational_age_category <span class="sc">+</span> prepreg_bmi_kgm2 <span class="sc">+</span> baby_birthweight_kg <span class="sc">+</span> baby_gender <span class="sc">+</span> mode_of_delivery <span class="sc">+</span> mother_age, <span class="at">data =</span> amc)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co">#model from DAG</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> moms_ed_level <span class="sc">+</span> SES_index_final <span class="sc">+</span> baby_gender, <span class="at">data =</span> amc)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>lm3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> SES_index_final <span class="sc">+</span> baby_gender, <span class="at">data =</span> amc)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ SES_index_final + breastmilk_per_day + 
    gestational_age_category + prepreg_bmi_kgm2 + baby_birthweight_kg + 
    baby_gender + mode_of_delivery + mother_age, data = amc)

Residuals:
    Min      1Q  Median      3Q     Max 
-45.087  -7.464  -0.009   6.649  43.544 

Coefficients:
                               Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                   121.65085    6.75986  17.996  &lt; 2e-16 ***
SES_index_final                -0.03500    0.03765  -0.930   0.3530    
breastmilk_per_day             -0.18510    0.13840  -1.337   0.1816    
gestational_age_category&lt;38    -2.81881    5.10242  -0.552   0.5809    
gestational_age_category&gt;42     6.24107    5.04598   1.237   0.2166    
gestational_age_category38-40   3.37608    4.32426   0.781   0.4353    
gestational_age_category40      4.96273    4.16482   1.192   0.2339    
gestational_age_category40-42   4.01301    4.23012   0.949   0.3432    
prepreg_bmi_kgm2               -0.54007    0.08486  -6.364 4.02e-10 ***
baby_birthweight_kg            -2.82276    1.18199  -2.388   0.0173 *  
baby_genderMale                -2.32755    0.97259  -2.393   0.0170 *  
mode_of_delivery               -5.89397    1.16258  -5.070 5.38e-07 ***
mother_age                      0.09402    0.07793   1.206   0.2281    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 11.31 on 576 degrees of freedom
  (107 observations deleted due to missingness)
Multiple R-squared:  0.151, Adjusted R-squared:  0.1333 
F-statistic: 8.534 on 12 and 576 DF,  p-value: 5.121e-15</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ moms_ed_level + SES_index_final + 
    baby_gender, data = amc)

Residuals:
    Min      1Q  Median      3Q     Max 
-54.607  -6.433   0.797   8.289  35.872 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     91.53725    1.56162  58.617  &lt; 2e-16 ***
moms_ed_level    1.60649    0.38496   4.173 3.46e-05 ***
SES_index_final -0.12429    0.04635  -2.682  0.00753 ** 
baby_genderMale -1.95237    0.99524  -1.962  0.05027 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 11.96 on 587 degrees of freedom
  (105 observations deleted due to missingness)
Multiple R-squared:  0.0336,    Adjusted R-squared:  0.02866 
F-statistic: 6.803 on 3 and 587 DF,  p-value: 0.0001635</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ SES_index_final + baby_gender, data = amc)

Residuals:
    Min      1Q  Median      3Q     Max 
-54.585  -4.810   0.372   6.788  36.981 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     95.16431    1.18804  80.102   &lt;2e-16 ***
SES_index_final -0.02145    0.03673  -0.584    0.559    
baby_genderMale -1.26627    0.93179  -1.359    0.175    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 12.11 on 684 degrees of freedom
  (9 observations deleted due to missingness)
Multiple R-squared:  0.003098,  Adjusted R-squared:  0.0001829 
F-statistic: 1.063 on 2 and 684 DF,  p-value: 0.3461</code></pre>
</div>
</div>
</section>
<section id="linear-regression-model-diagnostics-for-dag-models" class="level2">
<h2 class="anchored" data-anchor-id="linear-regression-model-diagnostics-for-dag-models">Linear Regression Model Diagnostics for DAG models</h2>
<p>http://sthda.com/english/articles/39-regression-model-diagnostics/161-linear-regression-assumptions-and-diagnostics-in-r-essentials</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-9-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-9-4.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>For the second (preferred DAG)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm2) <span class="co"># Should I remove 421 and 54?</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-10-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-10-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-10-4.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Simplest DAG</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-11-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-11-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-11-4.png" class="img-fluid" width="672"></p>
</div>
</div>
<hr>
</section>
<section id="no-dag-forward-model-selection-for-descriptive-variables" class="level2">
<h2 class="anchored" data-anchor-id="no-dag-forward-model-selection-for-descriptive-variables">No DAG: Forward model selection for descriptive variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>amc2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(amc, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(X,merge_id_dyad, dyad_id, bsid_notes, bsid_mot_cs, bsid_lang_cs,bsid_mot_fm_ss, cog_quartile,cog_mean_sd, bsid_mot_gm_ss,bsid_ab_cs,mot_low2high,bsid_se_cs,</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>    cog_low2high,bayleys_test_bsid3_complete,baby_gender_cat,gestational_age_cat))</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>amc3 <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(amc2)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce a model with many predictors </span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>m.full <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> ., <span class="at">data=</span>amc3, <span class="at">na.action =</span> <span class="st">"na.omit"</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce a model with no predictors</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>m.null <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data=</span>amc3, <span class="at">na.action =</span> <span class="st">"na.omit"</span>)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co">#forward selection model</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>m.forward <span class="ot">&lt;-</span> </span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step</span>(<span class="at">object =</span> m.null,</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">scope=</span><span class="fu">formula</span>(m.full), </span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">direction=</span><span class="st">"forward"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Start:  AIC=598.81
bsid_cog_cs ~ 1

                            Df Sum of Sq   RSS    AIC
+ prepreg_bmi_kgm2           1    382.43 13572 597.29
+ pets                       1    272.59 13681 598.31
+ faith_pd                   1    265.10 13689 598.38
&lt;none&gt;                                   13954 598.81
+ father_age                 1    193.41 13760 599.04
+ breastfeedings_continuous  1    163.05 13791 599.32
+ breastmilk_per_day         1    161.13 13793 599.34
+ gut_richness               1    160.22 13794 599.35
+ moms_ed_level              1     87.42 13866 600.02
+ mode_of_delivery           1     41.03 13913 600.44
+ mode_of_delivery_cat       1     41.03 13913 600.44
+ baby_gender                1     37.42 13916 600.47
+ baby_birthweight_kg        1     28.16 13926 600.56
+ mother_age                 1     18.54 13935 600.65
+ timepoint                  1     17.48 13936 600.66
+ SES_index_final            1     11.72 13942 600.71
+ shannon_entropy            1      1.40 13952 600.80
+ inf_antibiotics            1      0.01 13954 600.81
+ martial_status             1      0.00 13954 600.81
+ gestational_age_category   5    516.51 13437 604.02

Step:  AIC=597.29
bsid_cog_cs ~ prepreg_bmi_kgm2

                            Df Sum of Sq   RSS    AIC
+ pets                       1    362.57 13209 595.85
&lt;none&gt;                                   13572 597.29
+ faith_pd                   1    197.54 13374 597.42
+ father_age                 1    151.12 13420 597.86
+ breastfeedings_continuous  1    147.60 13424 597.90
+ breastmilk_per_day         1    145.25 13426 597.92
+ gut_richness               1    120.21 13451 598.16
+ moms_ed_level              1     47.93 13524 598.84
+ baby_gender                1     36.30 13535 598.95
+ mode_of_delivery_cat       1     27.89 13544 599.02
+ mode_of_delivery           1     27.89 13544 599.02
+ inf_antibiotics            1     25.71 13546 599.04
+ baby_birthweight_kg        1     18.79 13553 599.11
+ shannon_entropy            1     16.67 13555 599.13
+ timepoint                  1     13.85 13558 599.16
+ SES_index_final            1      7.02 13564 599.22
+ mother_age                 1      5.97 13566 599.23
+ martial_status             1      0.01 13572 599.29
+ gestational_age_category   5    499.68 13072 602.52

Step:  AIC=595.85
bsid_cog_cs ~ prepreg_bmi_kgm2 + pets

                            Df Sum of Sq   RSS    AIC
+ breastfeedings_continuous  1    249.09 12960 595.43
+ breastmilk_per_day         1    241.20 12968 595.51
&lt;none&gt;                                   13209 595.85
+ faith_pd                   1    205.45 13004 595.86
+ gut_richness               1    137.89 13071 596.51
+ father_age                 1    105.57 13103 596.83
+ baby_birthweight_kg        1     73.91 13135 597.13
+ mode_of_delivery           1     50.30 13159 597.36
+ mode_of_delivery_cat       1     50.30 13159 597.36
+ baby_gender                1     44.96 13164 597.41
+ moms_ed_level              1     20.61 13188 597.65
+ inf_antibiotics            1     12.08 13197 597.73
+ shannon_entropy            1      7.54 13201 597.77
+ timepoint                  1      7.39 13202 597.78
+ SES_index_final            1      7.19 13202 597.78
+ martial_status             1      2.42 13206 597.82
+ mother_age                 1      0.59 13208 597.84
+ gestational_age_category   5    403.29 12806 601.91

Step:  AIC=595.43
bsid_cog_cs ~ prepreg_bmi_kgm2 + pets + breastfeedings_continuous

                           Df Sum of Sq   RSS    AIC
&lt;none&gt;                                  12960 595.43
+ faith_pd                  1    134.15 12826 596.11
+ father_age                1    108.61 12851 596.36
+ gut_richness              1     87.90 12872 596.56
+ baby_gender               1     53.18 12907 596.91
+ baby_birthweight_kg       1     49.01 12911 596.95
+ moms_ed_level             1     30.83 12929 597.13
+ timepoint                 1     29.77 12930 597.14
+ shannon_entropy           1     20.80 12939 597.22
+ mode_of_delivery          1     14.14 12946 597.29
+ mode_of_delivery_cat      1     14.14 12946 597.29
+ SES_index_final           1     14.01 12946 597.29
+ breastmilk_per_day        1     12.12 12948 597.31
+ inf_antibiotics           1      5.61 12954 597.37
+ mother_age                1      3.57 12956 597.39
+ martial_status            1      2.24 12958 597.41
+ gestational_age_category  5    534.75 12425 600.08</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m.full)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ ., data = amc3, na.action = "na.omit")

Residuals:
    Min      1Q  Median      3Q     Max 
-19.362  -6.045  -0.939   5.748  29.288 

Coefficients: (1 not defined because of singularities)
                              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                   97.77081   17.19885   5.685 1.21e-07 ***
timepoint                      0.08041    0.43880   0.183   0.8550    
SES_index_final                0.09330    0.13757   0.678   0.4991    
moms_ed_level                 -0.39024    0.92834  -0.420   0.6751    
martial_status                 0.19201    0.50102   0.383   0.7023    
pets                           2.45309    2.31484   1.060   0.2917    
baby_genderMale               -0.86839    2.09586  -0.414   0.6795    
baby_birthweight_kg            0.95353    3.05798   0.312   0.7558    
prepreg_bmi_kgm2              -0.18424    0.18563  -0.993   0.3232    
mother_age                     0.28389    0.23083   1.230   0.2215    
father_age                    -0.45324    0.21675  -2.091   0.0390 *  
inf_antibiotics                3.39951    3.88514   0.875   0.3836    
breastmilk_per_day            -1.18503    4.14306  -0.286   0.7754    
gestational_age_category&lt;38   -5.03692   10.41545  -0.484   0.6297    
gestational_age_category&gt;42   16.33665   12.11173   1.349   0.1803    
gestational_age_category38-40 12.55428    7.35598   1.707   0.0909 .  
gestational_age_category40     6.03089    7.03763   0.857   0.3934    
gestational_age_category40-42  9.74479    7.05556   1.381   0.1702    
mode_of_delivery              -1.40370    2.74307  -0.512   0.6099    
breastfeedings_continuous      1.99416    5.08650   0.392   0.6958    
mode_of_delivery_catVaginal         NA         NA      NA       NA    
shannon_entropy                2.72633    1.67410   1.629   0.1064    
gut_richness                   0.10799    0.14185   0.761   0.4482    
faith_pd                      -2.25605    1.66262  -1.357   0.1777    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 10.38 on 104 degrees of freedom
Multiple R-squared:  0.1964,    Adjusted R-squared:  0.02644 
F-statistic: 1.156 on 22 and 104 DF,  p-value: 0.3042</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m.forward)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ prepreg_bmi_kgm2 + pets + breastfeedings_continuous, 
    data = amc3, na.action = "na.omit")

Residuals:
    Min      1Q  Median      3Q     Max 
-24.303  -6.743  -0.442   6.259  32.837 

Coefficients:
                          Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               102.2202     4.6282  22.086   &lt;2e-16 ***
prepreg_bmi_kgm2           -0.3393     0.1614  -2.102   0.0376 *  
pets                        4.2427     2.0216   2.099   0.0379 *  
breastfeedings_continuous   0.4568     0.2971   1.538   0.1267    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 10.26 on 123 degrees of freedom
Multiple R-squared:  0.07124,   Adjusted R-squared:  0.04859 
F-statistic: 3.145 on 3 and 123 DF,  p-value: 0.02765</code></pre>
</div>
</div>
</section>
<section id="forward-model-selection-for-alpha-diversity-variables" class="level2">
<h2 class="anchored" data-anchor-id="forward-model-selection-for-alpha-diversity-variables">Forward model selection for alpha diversity variables</h2>
<p>The residual sum of squares tells you how much of the dependent variable’s variation your model did not explain A lower RSS indicates a better fit of the regression model to the data.</p>
<p>The AIC (Akaike Information Criterion) is another measure of the quality of a statistical model. It is based on the difference between the maximum likelihood estimate of the model parameters and the number of parameters in the model. A lower AIC indicates a better fit of the model to the data.</p>
<p>You want both of these to be low.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#removing id so it's not included in the model</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>amc2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(amc, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(X,merge_id_dyad, dyad_id, bsid_notes, bsid_mot_cs, bsid_lang_cs,bsid_mot_fm_ss, cog_quartile,cog_mean_sd, bsid_mot_gm_ss,bsid_ab_cs,mot_low2high,bsid_se_cs,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    cog_low2high,bayleys_test_bsid3_complete,baby_gender_cat,gestational_age_cat))</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>amc3 <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(amc2)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>amcfaith <span class="ot">&lt;-</span> amc2 <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  na.omit <span class="sc">%&gt;%</span> <span class="fu">subset</span>(<span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(shannon_entropy))</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce a model with many predictors </span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>m.full_faith <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> ., <span class="at">data=</span>amcfaith, <span class="at">na.action =</span> <span class="st">"na.omit"</span>)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce a model with no predictors</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>m.null_faith <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> faith_pd, <span class="at">data=</span>amcfaith, <span class="at">na.action =</span> <span class="st">"na.omit"</span>)</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a><span class="co">#forward selection model</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>m.forward_faith <span class="ot">&lt;-</span> </span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step</span>(<span class="at">object =</span> m.null_faith,</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">scope=</span><span class="fu">formula</span>(m.full_faith), </span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">direction=</span><span class="st">"forward"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Start:  AIC=598.38
bsid_cog_cs ~ faith_pd

                            Df Sum of Sq   RSS    AIC
+ prepreg_bmi_kgm2           1    314.87 13374 597.42
+ pets                       1    288.56 13400 597.67
+ gut_richness               1    223.56 13465 598.29
&lt;none&gt;                                   13689 598.38
+ father_age                 1    176.83 13512 598.73
+ timepoint                  1    135.87 13553 599.11
+ breastfeedings_continuous  1     97.33 13592 599.47
+ breastmilk_per_day         1     94.80 13594 599.50
+ moms_ed_level              1     66.43 13622 599.76
+ mode_of_delivery           1     28.86 13660 600.11
+ mode_of_delivery_cat       1     28.86 13660 600.11
+ baby_gender                1     21.07 13668 600.18
+ mother_age                 1      7.82 13681 600.31
+ baby_birthweight_kg        1      6.75 13682 600.32
+ inf_antibiotics            1      2.68 13686 600.35
+ martial_status             1      1.17 13688 600.37
+ SES_index_final            1      0.57 13688 600.37
+ gestational_age_category   5    594.42 13094 602.74

Step:  AIC=597.42
bsid_cog_cs ~ faith_pd + prepreg_bmi_kgm2

                            Df Sum of Sq   RSS    AIC
+ pets                       1    370.48 13004 595.86
&lt;none&gt;                                   13374 597.42
+ gut_richness               1    163.89 13210 597.86
+ father_age                 1    141.63 13232 598.07
+ timepoint                  1    106.14 13268 598.41
+ breastfeedings_continuous  1     93.95 13280 598.53
+ breastmilk_per_day         1     91.22 13283 598.55
+ inf_antibiotics            1     38.21 13336 599.06
+ moms_ed_level              1     36.95 13337 599.07
+ baby_gender                1     22.17 13352 599.21
+ mode_of_delivery           1     20.04 13354 599.23
+ mode_of_delivery_cat       1     20.04 13354 599.23
+ baby_birthweight_kg        1      4.23 13370 599.38
+ mother_age                 1      1.64 13372 599.41
+ martial_status             1      0.82 13373 599.42
+ SES_index_final            1      0.15 13374 599.42
+ gestational_age_category   5    566.95 12807 601.92

Step:  AIC=595.86
bsid_cog_cs ~ faith_pd + prepreg_bmi_kgm2 + pets

                            Df Sum of Sq   RSS    AIC
&lt;none&gt;                                   13004 595.86
+ breastfeedings_continuous  1    177.79 12826 596.11
+ breastmilk_per_day         1    169.95 12834 596.19
+ gut_richness               1    111.14 12892 596.77
+ father_age                 1     97.00 12906 596.90
+ timepoint                  1     86.96 12916 597.00
+ baby_birthweight_kg        1     39.96 12964 597.46
+ mode_of_delivery           1     39.57 12964 597.47
+ mode_of_delivery_cat       1     39.57 12964 597.47
+ baby_gender                1     28.85 12975 597.57
+ inf_antibiotics            1     21.05 12982 597.65
+ moms_ed_level              1     13.25 12990 597.73
+ martial_status             1      0.33 13003 597.85
+ mother_age                 1      0.20 13003 597.85
+ SES_index_final            1      0.14 13003 597.85
+ gestational_age_category   5    453.35 12550 601.35</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m.full_faith)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ ., data = amcfaith, na.action = "na.omit")

Residuals:
     Min       1Q   Median       3Q      Max 
-19.8117  -6.3572  -0.9897   5.9284  31.1779 

Coefficients: (1 not defined because of singularities)
                              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                   99.36986   17.30535   5.742 9.17e-08 ***
timepoint                      0.08559    0.44223   0.194   0.8469    
SES_index_final                0.10233    0.13854   0.739   0.4618    
moms_ed_level                 -0.32417    0.93472  -0.347   0.7294    
martial_status                 0.05191    0.49745   0.104   0.9171    
pets                           2.58438    2.33157   1.108   0.2702    
baby_genderMale               -0.88051    2.11227  -0.417   0.6776    
baby_birthweight_kg            1.20406    3.07804   0.391   0.6965    
prepreg_bmi_kgm2              -0.18526    0.18709  -0.990   0.3243    
mother_age                     0.31887    0.23163   1.377   0.1716    
father_age                    -0.44176    0.21834  -2.023   0.0456 *  
inf_antibiotics                1.19473    3.67016   0.326   0.7454    
breastmilk_per_day            -1.69778    4.16345  -0.408   0.6843    
gestational_age_category&lt;38   -0.90723   10.18120  -0.089   0.9292    
gestational_age_category&gt;42   19.56864   12.04166   1.625   0.1071    
gestational_age_category38-40 14.06810    7.35419   1.913   0.0585 .  
gestational_age_category40     7.94509    6.99315   1.136   0.2585    
gestational_age_category40-42 11.21502    7.05240   1.590   0.1148    
mode_of_delivery              -1.17034    2.76079  -0.424   0.6725    
breastfeedings_continuous      2.62572    5.11144   0.514   0.6085    
mode_of_delivery_catVaginal         NA         NA      NA       NA    
gut_richness                   0.15418    0.14007   1.101   0.2735    
faith_pd                      -2.31794    1.67521  -1.384   0.1694    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 10.46 on 105 degrees of freedom
Multiple R-squared:  0.1759,    Adjusted R-squared:  0.01113 
F-statistic: 1.068 on 21 and 105 DF,  p-value: 0.394</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m.null_faith)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ faith_pd, data = amcfaith, na.action = "na.omit")

Residuals:
    Min      1Q  Median      3Q     Max 
-26.122  -6.875   0.299   5.009  33.201 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  99.9864     2.8780  34.741   &lt;2e-16 ***
faith_pd     -0.4904     0.3152  -1.556    0.122    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 10.46 on 125 degrees of freedom
Multiple R-squared:  0.019, Adjusted R-squared:  0.01115 
F-statistic: 2.421 on 1 and 125 DF,  p-value: 0.1223</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m.forward_faith)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = bsid_cog_cs ~ faith_pd + prepreg_bmi_kgm2 + pets, 
    data = amcfaith, na.action = "na.omit")

Residuals:
     Min       1Q   Median       3Q      Max 
-22.7318  -7.5408   0.1388   5.8490  30.7381 

Coefficients:
                 Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)      107.0031     4.9876  21.454   &lt;2e-16 ***
faith_pd          -0.4348     0.3119  -1.394   0.1658    
prepreg_bmi_kgm2  -0.3154     0.1628  -1.937   0.0550 .  
pets               3.7310     1.9930   1.872   0.0636 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 10.28 on 123 degrees of freedom
Multiple R-squared:  0.06811,   Adjusted R-squared:  0.04538 
F-statistic: 2.997 on 3 and 123 DF,  p-value: 0.03338</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(<span class="fu">summary</span>(m.forward_faith))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    Estimate Std. Error   t value     Pr(&gt;|t|)
(Intercept)      107.0031370  4.9876201 21.453747 2.163223e-43
faith_pd          -0.4348426  0.3119315 -1.394033 1.658210e-01
prepreg_bmi_kgm2  -0.3154120  0.1628086 -1.937318 5.499902e-02
pets               3.7309518  1.9930383  1.871992 6.358267e-02</code></pre>
</div>
</div>
<p>AIC deals with the trade-off between the goodness of fit of the model and the simplicity of the model, which helps to avoid both overfitting and underfitting.</p>
</section>
<section id="linear-regression-model-diagnostics-for-forward-selection-model" class="level2">
<h2 class="anchored" data-anchor-id="linear-regression-model-diagnostics-for-forward-selection-model">Linear Regression Model Diagnostics for Forward Selection model</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>DAGfaith <span class="ot">&lt;-</span> <span class="fu">lm</span>(bsid_cog_cs <span class="sc">~</span> faith_pd <span class="sc">+</span> SES_index_final <span class="sc">+</span> baby_gender <span class="sc">+</span> moms_ed_level, <span class="at">data =</span> amc6)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(DAGfaith) <span class="co">#46, 81, 90, 98 </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(m.forward_faith)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-14-8.png" class="img-fluid" width="672"></p>
</div>
</div>
<hr>
</section>
<section id="alpha-diversity-6month-loop-using-dag-and-forward-selection-models" class="level2">
<h2 class="anchored" data-anchor-id="alpha-diversity-6month-loop-using-dag-and-forward-selection-models">Alpha Diversity (6month) Loop using DAG and Forward Selection Models</h2>
<p>Not pretty. But it works.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ReSet working directories ####</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="st">"/Volumes/IPHY/ADORLab/__Users/emye7956/MM/code_review/outputfiles/"</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>d3 <span class="ot">&lt;-</span> <span class="st">"/Volumes/IPHY/ADORLab/__Users/emye7956/MM/code_review/input_files/"</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>d4 <span class="ot">&lt;-</span> <span class="st">"/Volumes/IPHY/ADORLab/HEI Study/16S Data/MM 16S Processed Files for Mom and Infants 05Jan21/taxonomy/"</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Input dataframe with correct timepoint (and prior editing)</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>a_meta <span class="ot">&lt;-</span> amc <span class="co"># rerun with meta_alpha to include inclomplete cases. I saw very little difference. </span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot theme for graphs to come </span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>mytheme <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.5</span>, </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">colour =</span> <span class="st">"black"</span>), </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>                 <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>),</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>                 <span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">95</span>, .<span class="dv">95</span>),</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="st">"right"</span>, <span class="st">"top"</span>),</span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>                 <span class="at">legend.box.just =</span> <span class="st">"right"</span>,</span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">legend.margin =</span> <span class="fu">margin</span>(<span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># REGRESSION LOOP</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting up exposure to loop through</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>ExposureFactors <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="st">"gut_richness"</span>,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"shannon_entropy"</span>,</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"faith_pd"</span>)</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co"># setting up outcomes to loop through </span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>InfantFactors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"bsid_se_cs"</span>, <span class="st">"bsid_ab_cs"</span>, </span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"bsid_mot_cs"</span>, <span class="st">"bsid_cog_cs"</span>,</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"bsid_lang_cs"</span>, <span class="st">"bsid_mot_gm_ss"</span>, <span class="st">"bsid_mot_fm_ss"</span>)</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co"># loop components </span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>exposureNamesList <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="co">#for model 1</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>pVal1ExposureList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>beta1ExposureList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a>beta1SDList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a>beta1ScaledList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>CI1ExposureLowerList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a>CI1ExposureUpperList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a>CI1UpperScaledList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a>CI1LowerScaledList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a><span class="co">#for model 2</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a>pVal2ExposureList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a>beta2ExposureList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>beta2SDList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a>beta2ScaledList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a>CI2ExposureLowerList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a>CI2ExposureUpperList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a>CI2UpperScaledList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>CI2LowerScaledList <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a>Exposure0List <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a>Outcome0List <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true" tabindex="-1"></a>allExposures <span class="ot">&lt;-</span> ExposureFactors</span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true" tabindex="-1"></a>allpvals <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true" tabindex="-1"></a>adjpvals <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="dv">0</span>)</span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true" tabindex="-1"></a><span class="co">#Sets up a loop to run through all cognitive outcomes</span></span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(InfantFactors)))</span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb45-43"><a href="#cb45-43" aria-hidden="true" tabindex="-1"></a>  <span class="co">#For each cognitive outcome, we then loop through each of our exposures &amp; incrementally build a linear model for each exposure/outcome pair</span></span>
<span id="cb45-44"><a href="#cb45-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ExposureFactors))) </span>
<span id="cb45-45"><a href="#cb45-45" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-46"><a href="#cb45-46" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb45-47"><a href="#cb45-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-48"><a href="#cb45-48" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Choose the timepoint of interest (remember to change the loop if you change it)</span></span>
<span id="cb45-49"><a href="#cb45-49" aria-hidden="true" tabindex="-1"></a>    meta_6m <span class="ot">&lt;-</span> a_meta <span class="sc">%&gt;%</span> <span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="dv">6</span>) <span class="co">#old run without removing </span></span>
<span id="cb45-50"><a href="#cb45-50" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-51"><a href="#cb45-51" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create dateframe without NAs (include all variables to be adjusted for in model)</span></span>
<span id="cb45-52"><a href="#cb45-52" aria-hidden="true" tabindex="-1"></a>thisDataInstance <span class="ot">=</span> <span class="fu">na.omit</span>(<span class="fu">data.frame</span>(<span class="at">thisOutcome=</span>meta_6m[,<span class="fu">names</span>(meta_6m) <span class="sc">%in%</span> InfantFactors[j]], <span class="at">thisExposureFactor =</span> meta_6m[,<span class="fu">names</span>(meta_6m) <span class="sc">%in%</span> ExposureFactors[i]],</span>
<span id="cb45-53"><a href="#cb45-53" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">baby_gender=</span>meta_6m<span class="sc">$</span>baby_gender,</span>
<span id="cb45-54"><a href="#cb45-54" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">birthweight=</span>meta_6m<span class="sc">$</span>baby_birthweight_kg,</span>
<span id="cb45-55"><a href="#cb45-55" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">SES_index_model=</span>meta_6m<span class="sc">$</span>SES_index_final,</span>
<span id="cb45-56"><a href="#cb45-56" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">prepreg_bmi_kgm2=</span>meta_6m<span class="sc">$</span>prepreg_bmi_kgm2,</span>
<span id="cb45-57"><a href="#cb45-57" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">maternalage=</span>meta_6m<span class="sc">$</span>mother_age,</span>
<span id="cb45-58"><a href="#cb45-58" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">gestationalagebl=</span>meta_6m<span class="sc">$</span>gestational_age_category,</span>
<span id="cb45-59"><a href="#cb45-59" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">mode_of_delivery=</span>meta_6m<span class="sc">$</span>mode_of_delivery,</span>
<span id="cb45-60"><a href="#cb45-60" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">moms_ed_level =</span> meta_6m<span class="sc">$</span>moms_ed_level,</span>
<span id="cb45-61"><a href="#cb45-61" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pets =</span> meta_6m<span class="sc">$</span>pets))</span>
<span id="cb45-62"><a href="#cb45-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-63"><a href="#cb45-63" aria-hidden="true" tabindex="-1"></a>    <span class="co">#linear model for plot</span></span>
<span id="cb45-64"><a href="#cb45-64" aria-hidden="true" tabindex="-1"></a>    modelFormForPlot <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(InfantFactors[j],<span class="st">"~"</span>,ExposureFactors[i]))</span>
<span id="cb45-65"><a href="#cb45-65" aria-hidden="true" tabindex="-1"></a>    statModForPlot <span class="ot">&lt;-</span> <span class="fu">lm</span>(modelFormForPlot, <span class="at">data=</span>meta_6m)</span>
<span id="cb45-66"><a href="#cb45-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-67"><a href="#cb45-67" aria-hidden="true" tabindex="-1"></a>    <span class="co">#1) Model 1 from DAG</span></span>
<span id="cb45-68"><a href="#cb45-68" aria-hidden="true" tabindex="-1"></a>    modelForm1<span class="ot">=</span><span class="fu">as.formula</span>(<span class="fu">paste</span>(InfantFactors[j],<span class="st">"~"</span>,ExposureFactors[i],<span class="st">"+"</span>,</span>
<span id="cb45-69"><a href="#cb45-69" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"SES_index_final"</span>, <span class="st">"+"</span>, </span>
<span id="cb45-70"><a href="#cb45-70" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"as.factor"</span>,<span class="st">"("</span>,<span class="st">"baby_gender"</span>,<span class="st">")"</span>, <span class="st">"+"</span>,</span>
<span id="cb45-71"><a href="#cb45-71" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"as.factor"</span>, <span class="st">"("</span>, <span class="st">"moms_ed_level"</span>,<span class="st">")"</span>))</span>
<span id="cb45-72"><a href="#cb45-72" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-73"><a href="#cb45-73" aria-hidden="true" tabindex="-1"></a>    <span class="co">#2) Model 2 from forward model selection</span></span>
<span id="cb45-74"><a href="#cb45-74" aria-hidden="true" tabindex="-1"></a>    modelForm2<span class="ot">=</span><span class="fu">as.formula</span>(<span class="fu">paste</span>(InfantFactors[j],<span class="st">"~"</span>,ExposureFactors[i],<span class="st">"+"</span>,</span>
<span id="cb45-75"><a href="#cb45-75" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"prepreg_bmi_kgm2"</span>, <span class="st">"+"</span>,</span>
<span id="cb45-76"><a href="#cb45-76" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"as.factor"</span>,<span class="st">"("</span>,<span class="st">"pets"</span>,<span class="st">")"</span>,</span>
<span id="cb45-77"><a href="#cb45-77" aria-hidden="true" tabindex="-1"></a>                  </span>
<span id="cb45-78"><a href="#cb45-78" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"*"</span>,ExposureFactors[i]))</span>
<span id="cb45-79"><a href="#cb45-79" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-80"><a href="#cb45-80" aria-hidden="true" tabindex="-1"></a>    <span class="co">#prints model to be viewed in console as code runs</span></span>
<span id="cb45-81"><a href="#cb45-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(modelForm1); <span class="fu">print</span>(modelForm2)</span>
<span id="cb45-82"><a href="#cb45-82" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-83"><a href="#cb45-83" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Save Beta and Pvalue and put into a running list</span></span>
<span id="cb45-84"><a href="#cb45-84" aria-hidden="true" tabindex="-1"></a>    statMod1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(modelForm1,<span class="at">data=</span>meta_6m) </span>
<span id="cb45-85"><a href="#cb45-85" aria-hidden="true" tabindex="-1"></a>    beta1Exposure <span class="ot">&lt;-</span> <span class="fu">summary</span>(statMod1)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>] <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-86"><a href="#cb45-86" aria-hidden="true" tabindex="-1"></a>    pVal1Exposure <span class="ot">&lt;-</span> <span class="fu">summary</span>(statMod1)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>] <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-87"><a href="#cb45-87" aria-hidden="true" tabindex="-1"></a>    beta1ExposureList <span class="ot">&lt;-</span> beta1ExposureList <span class="sc">%&gt;%</span> <span class="fu">append</span>(beta1Exposure) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-88"><a href="#cb45-88" aria-hidden="true" tabindex="-1"></a>    pVal1ExposureList <span class="ot">&lt;-</span> pVal1ExposureList <span class="sc">%&gt;%</span> <span class="fu">append</span>(pVal1Exposure) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-89"><a href="#cb45-89" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-90"><a href="#cb45-90" aria-hidden="true" tabindex="-1"></a>    CI1ExposureLower <span class="ot">&lt;-</span> <span class="fu">confint</span>(statMod1, <span class="at">level=</span><span class="fl">0.95</span>)[<span class="dv">2</span>,<span class="dv">1</span>] </span>
<span id="cb45-91"><a href="#cb45-91" aria-hidden="true" tabindex="-1"></a>    CI1ExposureUpper <span class="ot">&lt;-</span> <span class="fu">confint</span>(statMod1, <span class="at">level=</span><span class="fl">0.95</span>)[<span class="dv">2</span>,<span class="dv">2</span>] </span>
<span id="cb45-92"><a href="#cb45-92" aria-hidden="true" tabindex="-1"></a>    CI1ExposureLowerList <span class="ot">&lt;-</span> CI1ExposureLowerList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI1ExposureLower) </span>
<span id="cb45-93"><a href="#cb45-93" aria-hidden="true" tabindex="-1"></a>    CI1ExposureUpperList <span class="ot">&lt;-</span> CI1ExposureUpperList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI1ExposureUpper) </span>
<span id="cb45-94"><a href="#cb45-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-95"><a href="#cb45-95" aria-hidden="true" tabindex="-1"></a>    Exposure0List <span class="ot">&lt;-</span> Exposure0List <span class="sc">%&gt;%</span> <span class="fu">append</span>(InfantFactors[j])</span>
<span id="cb45-96"><a href="#cb45-96" aria-hidden="true" tabindex="-1"></a>    Outcome0List <span class="ot">&lt;-</span> Outcome0List <span class="sc">%&gt;%</span>  <span class="fu">append</span>(ExposureFactors[i])</span>
<span id="cb45-97"><a href="#cb45-97" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-98"><a href="#cb45-98" aria-hidden="true" tabindex="-1"></a>    <span class="co"># pulling SD from 6m data set </span></span>
<span id="cb45-99"><a href="#cb45-99" aria-hidden="true" tabindex="-1"></a>    beta1SD <span class="ot">&lt;-</span> <span class="fu">sd</span>(<span class="fu">as.numeric</span>((meta_6m[[ExposureFactors[i]]]) <span class="sc">%&gt;%</span> na.omit))</span>
<span id="cb45-100"><a href="#cb45-100" aria-hidden="true" tabindex="-1"></a>    beta1SDList <span class="ot">&lt;-</span> beta1SDList <span class="sc">%&gt;%</span> <span class="fu">append</span>(beta1SD)</span>
<span id="cb45-101"><a href="#cb45-101" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-102"><a href="#cb45-102" aria-hidden="true" tabindex="-1"></a>    beta1Scaled <span class="ot">&lt;-</span> (beta1Exposure <span class="sc">*</span> beta1SD) </span>
<span id="cb45-103"><a href="#cb45-103" aria-hidden="true" tabindex="-1"></a>    beta1ScaledList <span class="ot">&lt;-</span> beta1ScaledList <span class="sc">%&gt;%</span> <span class="fu">append</span>(beta1Scaled <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>)) </span>
<span id="cb45-104"><a href="#cb45-104" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-105"><a href="#cb45-105" aria-hidden="true" tabindex="-1"></a>    CI1UpperScaled <span class="ot">&lt;-</span> (CI1ExposureUpper <span class="sc">*</span> beta1SD) </span>
<span id="cb45-106"><a href="#cb45-106" aria-hidden="true" tabindex="-1"></a>    CI1UpperScaledList <span class="ot">&lt;-</span> CI1UpperScaledList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI1UpperScaled <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>))</span>
<span id="cb45-107"><a href="#cb45-107" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-108"><a href="#cb45-108" aria-hidden="true" tabindex="-1"></a>    CI1LowerScaled <span class="ot">&lt;-</span> (CI1ExposureLower <span class="sc">*</span> beta1SD) </span>
<span id="cb45-109"><a href="#cb45-109" aria-hidden="true" tabindex="-1"></a>    CI1LowerScaledList <span class="ot">&lt;-</span> CI1LowerScaledList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI1LowerScaled <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>))</span>
<span id="cb45-110"><a href="#cb45-110" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-111"><a href="#cb45-111" aria-hidden="true" tabindex="-1"></a>    <span class="co"># repeating for model 2</span></span>
<span id="cb45-112"><a href="#cb45-112" aria-hidden="true" tabindex="-1"></a>    statMod2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(modelForm2,<span class="at">data=</span>meta_6m)</span>
<span id="cb45-113"><a href="#cb45-113" aria-hidden="true" tabindex="-1"></a>    beta2Exposure <span class="ot">&lt;-</span> <span class="fu">summary</span>(statMod2)<span class="sc">$</span>coefficients[<span class="dv">5</span>,<span class="dv">1</span>] <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-114"><a href="#cb45-114" aria-hidden="true" tabindex="-1"></a>    pVal2Exposure <span class="ot">&lt;-</span> <span class="fu">summary</span>(statMod2)<span class="sc">$</span>coefficients[<span class="dv">5</span>,<span class="dv">4</span>] <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-115"><a href="#cb45-115" aria-hidden="true" tabindex="-1"></a>    beta2ExposureList <span class="ot">&lt;-</span> beta2ExposureList <span class="sc">%&gt;%</span> <span class="fu">append</span>(beta2Exposure <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>))</span>
<span id="cb45-116"><a href="#cb45-116" aria-hidden="true" tabindex="-1"></a>    pVal2ExposureList <span class="ot">&lt;-</span> pVal2ExposureList <span class="sc">%&gt;%</span> <span class="fu">append</span>(pVal2Exposure) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>)</span>
<span id="cb45-117"><a href="#cb45-117" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-118"><a href="#cb45-118" aria-hidden="true" tabindex="-1"></a>    CI2ExposureLower <span class="ot">&lt;-</span> <span class="fu">confint</span>(statMod2, <span class="at">level=</span><span class="fl">0.95</span>)[<span class="dv">2</span>,<span class="dv">1</span>] </span>
<span id="cb45-119"><a href="#cb45-119" aria-hidden="true" tabindex="-1"></a>    CI2ExposureUpper <span class="ot">&lt;-</span> <span class="fu">confint</span>(statMod2, <span class="at">level=</span><span class="fl">0.95</span>)[<span class="dv">2</span>,<span class="dv">2</span>] </span>
<span id="cb45-120"><a href="#cb45-120" aria-hidden="true" tabindex="-1"></a>    CI2ExposureLowerList <span class="ot">&lt;-</span> CI2ExposureLowerList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI2ExposureLower) </span>
<span id="cb45-121"><a href="#cb45-121" aria-hidden="true" tabindex="-1"></a>    CI2ExposureUpperList <span class="ot">&lt;-</span> CI2ExposureUpperList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI2ExposureUpper) </span>
<span id="cb45-122"><a href="#cb45-122" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-123"><a href="#cb45-123" aria-hidden="true" tabindex="-1"></a>    <span class="co"># repeating for model 2. pulling SD from 6m data set  </span></span>
<span id="cb45-124"><a href="#cb45-124" aria-hidden="true" tabindex="-1"></a>    beta2SD <span class="ot">&lt;-</span> <span class="fu">sd</span>(<span class="fu">as.numeric</span>((meta_6m[[ExposureFactors[i]]]) <span class="sc">%&gt;%</span> na.omit))</span>
<span id="cb45-125"><a href="#cb45-125" aria-hidden="true" tabindex="-1"></a>    beta2SDList <span class="ot">&lt;-</span> beta2SDList <span class="sc">%&gt;%</span> <span class="fu">append</span>(beta2SD)</span>
<span id="cb45-126"><a href="#cb45-126" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-127"><a href="#cb45-127" aria-hidden="true" tabindex="-1"></a>    beta2Scaled <span class="ot">&lt;-</span> (beta2Exposure <span class="sc">*</span> beta2SD) </span>
<span id="cb45-128"><a href="#cb45-128" aria-hidden="true" tabindex="-1"></a>    beta2ScaledList <span class="ot">&lt;-</span> beta2ScaledList <span class="sc">%&gt;%</span> <span class="fu">append</span>(beta2Scaled <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>)) </span>
<span id="cb45-129"><a href="#cb45-129" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-130"><a href="#cb45-130" aria-hidden="true" tabindex="-1"></a>    CI2UpperScaled <span class="ot">&lt;-</span> (CI2ExposureUpper <span class="sc">*</span> beta2SD) </span>
<span id="cb45-131"><a href="#cb45-131" aria-hidden="true" tabindex="-1"></a>    CI2UpperScaledList <span class="ot">&lt;-</span> CI2UpperScaledList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI2UpperScaled <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>))</span>
<span id="cb45-132"><a href="#cb45-132" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-133"><a href="#cb45-133" aria-hidden="true" tabindex="-1"></a>    CI2LowerScaled <span class="ot">&lt;-</span> (CI2ExposureLower <span class="sc">*</span> beta2SD) </span>
<span id="cb45-134"><a href="#cb45-134" aria-hidden="true" tabindex="-1"></a>    CI2LowerScaledList <span class="ot">&lt;-</span> CI2LowerScaledList <span class="sc">%&gt;%</span> <span class="fu">append</span>(CI2LowerScaled <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>))</span>
<span id="cb45-135"><a href="#cb45-135" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-136"><a href="#cb45-136" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Plots show beta and p value for statMod1</span></span>
<span id="cb45-137"><a href="#cb45-137" aria-hidden="true" tabindex="-1"></a>    title <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"Effect Size = "</span>,</span>
<span id="cb45-138"><a href="#cb45-138" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">format</span>(beta1Exposure,<span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb45-139"><a href="#cb45-139" aria-hidden="true" tabindex="-1"></a>                   <span class="st">", p-value="</span>,</span>
<span id="cb45-140"><a href="#cb45-140" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">format</span>(pVal1Exposure,<span class="at">digits=</span><span class="dv">3</span>), </span>
<span id="cb45-141"><a href="#cb45-141" aria-hidden="true" tabindex="-1"></a>                   <span class="st">", n = "</span>,<span class="fu">nrow</span>(thisDataInstance), <span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb45-142"><a href="#cb45-142" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-143"><a href="#cb45-143" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">qplot</span>(thisExposureFactor,thisOutcome, <span class="at">data =</span> thisDataInstance) <span class="sc">+</span></span>
<span id="cb45-144"><a href="#cb45-144" aria-hidden="true" tabindex="-1"></a>            <span class="fu">ggtitle</span>(title) <span class="sc">+</span> </span>
<span id="cb45-145"><a href="#cb45-145" aria-hidden="true" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">x=</span>ExposureFactors[i], <span class="at">y=</span>InfantFactors[j]) <span class="sc">+</span> </span>
<span id="cb45-146"><a href="#cb45-146" aria-hidden="true" tabindex="-1"></a>            mytheme <span class="sc">+</span> </span>
<span id="cb45-147"><a href="#cb45-147" aria-hidden="true" tabindex="-1"></a>            <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">'lm'</span>))</span>
<span id="cb45-148"><a href="#cb45-148" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb45-149"><a href="#cb45-149" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-150"><a href="#cb45-150" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_se_cs ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_se_cs ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `qplot()` was deprecated in ggplot2 3.4.0.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_se_cs ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_se_cs ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_se_cs ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_se_cs ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_ab_cs ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_ab_cs ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_ab_cs ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_ab_cs ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_ab_cs ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_ab_cs ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_cs ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_cs ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_cs ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_cs ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_cs ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_cs ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_cog_cs ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_cog_cs ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_cog_cs ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_cog_cs ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_cog_cs ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_cog_cs ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-12.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_lang_cs ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_lang_cs ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-13.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_lang_cs ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_lang_cs ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-14.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_lang_cs ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_lang_cs ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-15.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_gm_ss ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_gm_ss ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-16.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_gm_ss ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_gm_ss ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-17.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_gm_ss ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_gm_ss ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-18.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_fm_ss ~ gut_richness + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_fm_ss ~ gut_richness + prepreg_bmi_kgm2 + as.factor(pets) * 
    gut_richness</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-19.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_fm_ss ~ shannon_entropy + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_fm_ss ~ shannon_entropy + prepreg_bmi_kgm2 + as.factor(pets) * 
    shannon_entropy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-20.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>bsid_mot_fm_ss ~ faith_pd + SES_index_final + as.factor(baby_gender) + 
    as.factor(moms_ed_level)
bsid_mot_fm_ss ~ faith_pd + prepreg_bmi_kgm2 + as.factor(pets) * 
    faith_pd</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Lab-meeting-Put-together_files/figure-html/unnamed-chunk-15-21.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">graphics.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="table-alpha-diversity-regression-loop-outcomes" class="level2">
<h2 class="anchored" data-anchor-id="table-alpha-diversity-regression-loop-outcomes">Table alpha diversity regression loop outcomes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Table of model outcomes ####</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>LM1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(Exposure0List, Outcome0List, pVal1ExposureList, beta1ExposureList, CI1ExposureLowerList, CI1ExposureUpperList) <span class="sc">%&gt;%</span> </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>LM2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(Exposure0List, Outcome0List, pVal2ExposureList, beta2ExposureList, CI2ExposureLowerList, CI2ExposureUpperList) <span class="sc">%&gt;%</span> </span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>LM1scaled <span class="ot">&lt;-</span> <span class="fu">cbind</span>(Exposure0List, Outcome0List, pVal1ExposureList, beta1ScaledList, CI1LowerScaledList, CI1UpperScaledList) <span class="sc">%&gt;%</span> </span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>LM2scaled <span class="ot">&lt;-</span> <span class="fu">cbind</span>(Exposure0List, Outcome0List, pVal2ExposureList, beta2ExposureList, CI2LowerScaledList, CI2UpperScaledList) <span class="sc">%&gt;%</span> </span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>LM1scaled <span class="sc">%&gt;%</span> <span class="fu">kable</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_classic</span>(<span class="at">full_width =</span> F) <span class="co">#DAG model output</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> Exposure0List </th>
   <th style="text-align:left;"> Outcome0List </th>
   <th style="text-align:left;"> pVal1ExposureList </th>
   <th style="text-align:left;"> beta1ScaledList </th>
   <th style="text-align:left;"> CI1LowerScaledList </th>
   <th style="text-align:left;"> CI1UpperScaledList </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> bsid_se_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.179 </td>
   <td style="text-align:left;"> -2.37 </td>
   <td style="text-align:left;"> -5.81 </td>
   <td style="text-align:left;"> 1.1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_se_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.769 </td>
   <td style="text-align:left;"> -0.51 </td>
   <td style="text-align:left;"> -3.95 </td>
   <td style="text-align:left;"> 2.93 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_se_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.066 </td>
   <td style="text-align:left;"> -3.22 </td>
   <td style="text-align:left;"> -6.66 </td>
   <td style="text-align:left;"> 0.21 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_ab_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.372 </td>
   <td style="text-align:left;"> -1.79 </td>
   <td style="text-align:left;"> -5.73 </td>
   <td style="text-align:left;"> 2.17 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_ab_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.972 </td>
   <td style="text-align:left;"> -0.07 </td>
   <td style="text-align:left;"> -3.98 </td>
   <td style="text-align:left;"> 3.84 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_ab_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.253 </td>
   <td style="text-align:left;"> -2.28 </td>
   <td style="text-align:left;"> -6.23 </td>
   <td style="text-align:left;"> 1.67 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.013 </td>
   <td style="text-align:left;"> -3.25 </td>
   <td style="text-align:left;"> -5.79 </td>
   <td style="text-align:left;"> -0.69 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.824 </td>
   <td style="text-align:left;"> -0.29 </td>
   <td style="text-align:left;"> -2.9 </td>
   <td style="text-align:left;"> 2.31 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.007 </td>
   <td style="text-align:left;"> -3.53 </td>
   <td style="text-align:left;"> -6.07 </td>
   <td style="text-align:left;"> -0.99 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_cog_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.052 </td>
   <td style="text-align:left;"> -2.56 </td>
   <td style="text-align:left;"> -5.12 </td>
   <td style="text-align:left;"> 0.02 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_cog_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.702 </td>
   <td style="text-align:left;"> -0.5 </td>
   <td style="text-align:left;"> -3.08 </td>
   <td style="text-align:left;"> 2.09 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_cog_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.04 </td>
   <td style="text-align:left;"> -2.7 </td>
   <td style="text-align:left;"> -5.27 </td>
   <td style="text-align:left;"> -0.13 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_lang_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.148 </td>
   <td style="text-align:left;"> -1.73 </td>
   <td style="text-align:left;"> -4.08 </td>
   <td style="text-align:left;"> 0.63 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_lang_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.336 </td>
   <td style="text-align:left;"> -1.14 </td>
   <td style="text-align:left;"> -3.47 </td>
   <td style="text-align:left;"> 1.2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_lang_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.086 </td>
   <td style="text-align:left;"> -2.05 </td>
   <td style="text-align:left;"> -4.4 </td>
   <td style="text-align:left;"> 0.3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_gm_ss </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.107 </td>
   <td style="text-align:left;"> -0.44 </td>
   <td style="text-align:left;"> -0.97 </td>
   <td style="text-align:left;"> 0.1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_gm_ss </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.549 </td>
   <td style="text-align:left;"> -0.16 </td>
   <td style="text-align:left;"> -0.7 </td>
   <td style="text-align:left;"> 0.37 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_gm_ss </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.09 </td>
   <td style="text-align:left;"> -0.46 </td>
   <td style="text-align:left;"> -1 </td>
   <td style="text-align:left;"> 0.07 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_fm_ss </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.007 </td>
   <td style="text-align:left;"> -0.63 </td>
   <td style="text-align:left;"> -1.1 </td>
   <td style="text-align:left;"> -0.18 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_fm_ss </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.833 </td>
   <td style="text-align:left;"> 0.05 </td>
   <td style="text-align:left;"> -0.42 </td>
   <td style="text-align:left;"> 0.52 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_fm_ss </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.003 </td>
   <td style="text-align:left;"> -0.71 </td>
   <td style="text-align:left;"> -1.16 </td>
   <td style="text-align:left;"> -0.25 </td>
  </tr>
</tbody>
</table>

</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>LM2scaled <span class="sc">%&gt;%</span> <span class="fu">kable</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_classic</span>(<span class="at">full_width =</span> F) <span class="co">#forward model selection output </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> Exposure0List </th>
   <th style="text-align:left;"> Outcome0List </th>
   <th style="text-align:left;"> pVal2ExposureList </th>
   <th style="text-align:left;"> beta2ExposureList </th>
   <th style="text-align:left;"> CI2LowerScaledList </th>
   <th style="text-align:left;"> CI2UpperScaledList </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> bsid_se_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.447 </td>
   <td style="text-align:left;"> -0.17 </td>
   <td style="text-align:left;"> -4.63 </td>
   <td style="text-align:left;"> 2.58 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_se_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.77 </td>
   <td style="text-align:left;"> -1.4 </td>
   <td style="text-align:left;"> -3.67 </td>
   <td style="text-align:left;"> 4.7 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_se_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.289 </td>
   <td style="text-align:left;"> -2.5 </td>
   <td style="text-align:left;"> -5.34 </td>
   <td style="text-align:left;"> 2.04 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_ab_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.025 </td>
   <td style="text-align:left;"> -0.57 </td>
   <td style="text-align:left;"> -4.01 </td>
   <td style="text-align:left;"> 3.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_ab_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.084 </td>
   <td style="text-align:left;"> -9.07 </td>
   <td style="text-align:left;"> -1.94 </td>
   <td style="text-align:left;"> 7.21 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_ab_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.007 </td>
   <td style="text-align:left;"> -7 </td>
   <td style="text-align:left;"> -4.09 </td>
   <td style="text-align:left;"> 3.86 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.021 </td>
   <td style="text-align:left;"> -0.39 </td>
   <td style="text-align:left;"> -4.91 </td>
   <td style="text-align:left;"> 0.3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.282 </td>
   <td style="text-align:left;"> -3.91 </td>
   <td style="text-align:left;"> -2.53 </td>
   <td style="text-align:left;"> 3.83 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.102 </td>
   <td style="text-align:left;"> -2.85 </td>
   <td style="text-align:left;"> -5.42 </td>
   <td style="text-align:left;"> 0.02 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_cog_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.371 </td>
   <td style="text-align:left;"> -0.15 </td>
   <td style="text-align:left;"> -3.57 </td>
   <td style="text-align:left;"> 1.58 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_cog_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.939 </td>
   <td style="text-align:left;"> 0.26 </td>
   <td style="text-align:left;"> -2.5 </td>
   <td style="text-align:left;"> 3.5 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_cog_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.805 </td>
   <td style="text-align:left;"> -0.42 </td>
   <td style="text-align:left;"> -4.05 </td>
   <td style="text-align:left;"> 1.28 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_lang_cs </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.155 </td>
   <td style="text-align:left;"> -0.21 </td>
   <td style="text-align:left;"> -2.7 </td>
   <td style="text-align:left;"> 1.89 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_lang_cs </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.413 </td>
   <td style="text-align:left;"> -2.5 </td>
   <td style="text-align:left;"> -2.41 </td>
   <td style="text-align:left;"> 2.95 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_lang_cs </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.203 </td>
   <td style="text-align:left;"> -1.93 </td>
   <td style="text-align:left;"> -2.96 </td>
   <td style="text-align:left;"> 1.79 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_gm_ss </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.014 </td>
   <td style="text-align:left;"> -0.09 </td>
   <td style="text-align:left;"> -0.69 </td>
   <td style="text-align:left;"> 0.38 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_gm_ss </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.167 </td>
   <td style="text-align:left;"> -1.01 </td>
   <td style="text-align:left;"> -0.47 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_gm_ss </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.047 </td>
   <td style="text-align:left;"> -0.71 </td>
   <td style="text-align:left;"> -0.74 </td>
   <td style="text-align:left;"> 0.38 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_fm_ss </td>
   <td style="text-align:left;"> gut_richness </td>
   <td style="text-align:left;"> 0.163 </td>
   <td style="text-align:left;"> -0.04 </td>
   <td style="text-align:left;"> -1.09 </td>
   <td style="text-align:left;"> -0.12 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_fm_ss </td>
   <td style="text-align:left;"> shannon_entropy </td>
   <td style="text-align:left;"> 0.666 </td>
   <td style="text-align:left;"> -0.29 </td>
   <td style="text-align:left;"> -0.56 </td>
   <td style="text-align:left;"> 0.63 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> bsid_mot_fm_ss </td>
   <td style="text-align:left;"> faith_pd </td>
   <td style="text-align:left;"> 0.474 </td>
   <td style="text-align:left;"> -0.23 </td>
   <td style="text-align:left;"> -1.22 </td>
   <td style="text-align:left;"> -0.21 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<hr>
</section>
<section id="try-out-generalized-additive-models-gam" class="level2">
<h2 class="anchored" data-anchor-id="try-out-generalized-additive-models-gam">Try out Generalized Additive Models (GAM)</h2>
<p>https://multithreaded.stitchfix.com/blog/2015/07/30/gam/</p>
<div class="cell">

</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>